scRNA-seq-advanced/05-aucell.Rmd

```diff
@@ -0,0 +1,403 @@
+
+We will use an snRNA-seq of a Ewing sarcoma sample from the [`SCPCP000015` project](https://scpca.alexslemonade.org/projects/SCPCP000015) on the Single-cell Pediatric Cancer Atlas Portal and two relevant gene sets from the Molecular Signatures Database (MSigDB) to demonstrate this method.
+
+## Set up
+
+### Libraries
+
+```{r libraries}
+# We will be loading a SingleCellExperiment object into our environment but 
+# don't need to see the startup messages
+suppressPackageStartupMessages({
+  library(SingleCellExperiment)
+})
+
+# Library we'll use for the gene set analysis itself
+library(AUCell)
+
+# Libraries for accessing and working with gene sets
+library(GSEABase)
+library(msigdbr)
+```
+### Directories and files
+
+#### Directories
+
+```{r setup_directories}
+# Input data 
+ewing_data_dir <- fs::path("data", "ewing-sarcoma")
+processed_dir <- fs::path(ewing_data_dir, "processed")
+
+# Directory for holding pathway analysis results
+analysis_dir <- fs::path("analysis", "ewing-sarcoma", "pathway-analysis")
+# Create if it doesn't exist yet
+fs::dir_create(analysis_dir)
+```
+
+#### Files
+
+The input will be a `SingleCellExperiment` for an individual Ewing sarcoma library.
+
+```{r setup_input_files}
+sce_file <- fs::path(processed_dir, 
+                     "SCPCS000490", 
+                     "SCPCL000822_processed.rds")
+```
+
+We will save the AUCell results as a table in the analysis directory.
+
+```{r setup_output_files, live = TRUE}
+output_file <- fs::path(analysis_dir,
+                        "ewing_sarcoma_aucell_results.tsv")
+```
+
+
+### Functions
+
+The `source()` function allows us to load in custom functions we saved in an `.R` file.
+
+```{r source_functions}
+source(fs::path("util", "aucell_functions.R"))
+```
+
+This loads one custom function, called `plot_recovery_curve()`, into our environment.
+This function is adapted from [the AUCell vignette](https://github.com/aertslab/AUCell/blob/91753b327a39dc7a4bbed46408ec2271c485f2f0/vignettes/AUCell.Rmd#L295-L316).
+
+## Set up gene sets
+
+We are going to use two gene sets pertaining to Ewing sarcoma.
+
+* [`ZHANG_TARGETS_OF_EWSR1_FLI1_FUSION`](https://www.gsea-msigdb.org/gsea/msigdb/geneset_page.jsp?geneSetName=ZHANG_TARGETS_OF_EWSR1_FLI1_FUSION), which are genes that were highly expressed in a rhabdomyosarcoma cell line engineered to express the EWSR1-FLI1 fusion.
+* [`RIGGI_EWING_SARCOMA_PROGENITOR_UP`](https://www.gsea-msigdb.org/gsea/msigdb/cards/RIGGI_EWING_SARCOMA_PROGENITOR_UP), which are genes that were highly expressed in mesenchymal stem cells engineered to express the EWS-FLI1 fusion protein.
+
+We would expect both of these gene sets to have high expression in tumor cells.
+
+```{r genesets}
+ewing_gene_set_names <- c("ZHANG_TARGETS_OF_EWSR1_FLI1_FUSION",
+                          "RIGGI_EWING_SARCOMA_PROGENITOR_UP")
```

Comment:
I might make this a named list so that we can avoid all the `ewing_gene_set_names[1]` and use `ewing_gene_set_names["zhang"]`. Or I think you could rename the `msigdbr` results to have shorter names so you can easily use them in later code.